flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/CodeGeneratorContext.scala

```diff
@@ -671,28 +677,33 @@ class CodeGeneratorContext(val tableConfig: TableConfig) {
   /**
    * Adds a reusable [[DataStructureConverter]] to the member area of the generated class.
    *
-   * @param converter converter to be added
+   * @param dataType converter to be added
    * @param classLoaderTerm term to access the [[ClassLoader]] for user-defined classes
    */
   def addReusableConverter(
-      converter: DataStructureConverter[_, _],
+      dataType: DataType,
       classLoaderTerm: String = null)
     : String = {
+    reusableConverters.get(dataType) match {
```

Comment:
nit: should we ignore the nullability? If I am not mistaken we use the same converters for both nullable and not null types. In `DataTypeConverters` we use (`LogicalTypeRoot`, `conversionClass`) as a composite key.

Reply:
The mapping `DataTypeConverters` maps to a factory. But the factory considers nullability. This is why FLINK-18701 was discovered for example.

Reply:
My bad. You are right.